src/sentry/api/endpoints/system_options.py

```diff
@@ -59,6 +59,12 @@ def put(self, request):
                         'option': k,
                     },
                 }, status=400)
+
+            try:
+                if not (option.flags & options.FLAG_ALLOW_EMPTY) and not v:
```

Comment:
This is done because if the value is allowed to be falsey, we don't want to be deleting it.